extensions/common/aws/aws-s3-core/src/test/java/org/eclipse/edc/aws/s3/S3CoreExtensionTest.java

```diff
@@ -0,0 +1,98 @@
+ *
+ *  Contributors:
+ *       Masatake Iwasaki (NTT DATA) - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.aws.s3;
+
+import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.eclipse.edc.junit.testfixtures.MockVault;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(EdcExtension.class)
+public class S3CoreExtensionTest {
+    private static final String VAULT_KEY_ACCESS_KEY_ID = "key of access key id";
+    private static final String VAULT_KEY_SECRET_ACCESS_KEY = "key of secret access key";
+    private static final String EDC_AWS_ACCESS_KEY_ID = "access key id in vault";
+    private static final String EDC_AWS_SECRET_ACCESS_KEY = "secret access key in vault";
+    private static final String AWS_ACCESS_KEY_ID = "access key id via AWS SDK settings";
+    private static final String AWS_SECRET_ACCESS_KEY = "secret access key via AWS SDK settings";
+
+    private Properties savedProperties;
+    private S3CoreExtension s3;
+
+    @BeforeEach
+    void setUp(EdcExtension edc) {
+        savedProperties = (Properties) System.getProperties().clone();
+
+        // AWS SDK settings via system properties
+        System.setProperty("aws.accessKeyId", AWS_ACCESS_KEY_ID);
+        System.setProperty("aws.secretAccessKey", AWS_SECRET_ACCESS_KEY);
+
+        edc.setConfiguration(Map.of(
+                S3CoreExtension.AWS_ACCESS_KEY, VAULT_KEY_ACCESS_KEY_ID,
+                S3CoreExtension.AWS_SECRET_KEY, VAULT_KEY_SECRET_ACCESS_KEY));
+        edc.registerServiceMock(Vault.class, new MockVault());
+        s3 = new S3CoreExtension();
+        edc.registerSystemExtension(ServiceExtension.class, s3);
+    }
+
+    @AfterEach
+    void afterEach() {
+        System.setProperties(savedProperties);
+    }
+
+    @Test
+    void testCredentialsFromVault(EdcExtension edc, Vault vault) {
+        vault.storeSecret(VAULT_KEY_ACCESS_KEY_ID, EDC_AWS_ACCESS_KEY_ID);
+        vault.storeSecret(VAULT_KEY_SECRET_ACCESS_KEY, EDC_AWS_SECRET_ACCESS_KEY);
+
+        // EDC configurations have higher precedence than AWS SDK settings
+        var credentials = s3.createCredentialsProvider(edc.getContext()).resolveCredentials();
+        assertEquals(EDC_AWS_ACCESS_KEY_ID, credentials.accessKeyId());
+        assertEquals(EDC_AWS_SECRET_ACCESS_KEY, credentials.secretAccessKey());
+    }
+
+    @Test
+    void testCredentialsFromAwsSdk(EdcExtension edc) {
+        var credentials = s3.createCredentialsProvider(edc.getContext()).resolveCredentials();
+        assertEquals(AWS_ACCESS_KEY_ID, credentials.accessKeyId());
```

Comment:
I think it would be just enough to test that the provider is a `DefaultCredentialsProvider`, no real need to set the credentials in properties and then verify them because that's a behavior implemented by the aws sdk and we don't want to cover that.

Reply:
I updated the assertion based on your suggestion.